OptimizelySDK/Optimizely.cs

```diff
@@ -1093,13 +1380,52 @@ public OptimizelyConfig GetOptimizelyConfig()
                 return ((IOptimizelyConfigManager)ProjectConfigManager).GetOptimizelyConfig();
             }
 
-            Logger.Log(LogLevel.DEBUG, "ProjectConfigManager is not instance of IOptimizelyConfigManager, generating new OptimizelyConfigObject as a fallback");
+            Logger.Log(LogLevel.DEBUG,
+                "ProjectConfigManager is not instance of IOptimizelyConfigManager, generating new OptimizelyConfigObject as a fallback");
 
             return new OptimizelyConfigService(config).GetOptimizelyConfig();
         }
 
         #endregion FeatureFlag APIs
 
+#if USE_ODP
+        /// <summary>
+        /// Attempts to fetch and return a list of a user's qualified segments.
+        /// </summary>
+        /// <param name="userId">FS User ID</param>
+        /// <param name="segmentOptions">Options used during segment cache handling</param>
+        /// <returns>Qualified segments for the user from the cache or the ODP server</returns>
+        internal string[] FetchQualifiedSegments(string userId,
+            List<OdpSegmentOption> segmentOptions
+        )
+        {
+            return OdpManager?.FetchQualifiedSegments(userId, segmentOptions) ?? new string[0];
```

Comment:
We should return null as is since it should be differentiated as fetch error from empty segments (valid).
We may need more unit tests covering failed fetch from UserContext.